tink/internal/client_test/hardware_test.go

```diff
@@ -0,0 +1,123 @@
+/*
+Copyright 2020 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package client_test
+
+import (
+	"context"
+	"testing"
+
+	. "github.com/onsi/gomega"
+	"github.com/tinkerbell/cluster-api-provider-tinkerbell/tink/internal/client"
+)
+
+func TestHardwareLifecycle(t *testing.T) {
+	g := NewWithT(t)
+	ctx := context.Background()
+	hardwareClient := client.NewHardwareClient(realHardwareClient(t))
```

Comment:
Hm, we have a fake client for Tinkerbell, I guess we can use it here so we don't require real tinkerbell server?

BTW, if we want tests with Tinkerbell server, we can probably borrow the setup from https://github.com/tinkerbell/terraform-provider-tinkerbell/tree/master/test.